plateau.cpp

```diff
@@ -0,0 +1,116 @@
+#pragma warning(disable:4996)
+#include <cstdio>
+#include <iostream>
+#include <cstring>
+#include <cassert>
+#include <cstdlib>
+
+#include "fonction_conteneurs.h"
+#include "fonction_listes.h"
+
+#include "plateau.h"
+
+
+bool lire_plateau(Liste_mot& liste_mot, Plateau& plateau)
+{
+	Mot buffer;
+	bool compteur = false;
+
+	while (1)
+	{
+		scanf("%s", buffer);
+
+		if (strcmp(buffer, "*") == 0) {
+			return compteur; // si compteur = 0 on a entré seulement * dans la liste
+		}
+
+		if (!dans_le_plateau(plateau, buffer))
+			continue;
+
+		if (!exister(liste_mot, buffer)) {
+			ajouter(liste_mot, buffer);
+			compteur = true;
+		}
+	}
+
+	// inutile mais bon...
+	return compteur;
+}
+
+bool ajouter_plateau(Plateau& p)
+{
+	Mot unMot;
+	for (unsigned int i = 0; i < TAILLE_PLATEAU; i++)
+	{
+		scanf("%5s", unMot);
+		for (unsigned int j = 0; j < TAILLE_PLATEAU; j++)
+		{
+			p[i][j] = unMot[j];
+		}
+	}
+
+	return true;
+}
+
+void initialiser_plateau(Plateau_bool& p_bool)
```

Comment:
Bien séparer les fonctions relatives au type `Plateau` et `Plateau_bool` respectivement en début et fin de fichier
Elles sont mélangées et porte à confusion

Reply:
fixed sur 9a8fe90b. Il reste plus qu'à vérifier les in/out/ultime